### ISSUE TITLE
Payload generation optimization makes performance worse

### ISSUE THREAD

**Issue by jchambers** [relayrides/pushy#282]
Well, I goofed. I flipped the numbers on my benchmark results for the payload generation optimization introduced in #267, and the new approach is actually slower than the old approach. There are two things to do here:

1. Actually make payload generation faster.
2. Move to a more robust benchmarking strategy.

For the latter, I'll be looking to add [jmh](http://openjdk.java.net/projects/code-tools/jmh/) to the project.

**Comment by jchambers** [comment]
Well, this has certainly turned into quite the comedy of errors. Yes, I did flip the results of the previous benchmarks, but it turns out those benchmarks were bogus. Using the microbenchmarks introduced in #288, we can show that the changes introduced in #267 WERE a significant improvement:

### 0.6.1

| Benchmark                   | Mode  | Cnt | Score        | Error    | Units |
|-----------------------------|-------|-----|--------------|----------|-------|
| testLongAsciiMessageBody    | thrpt | 80  | 7.958 ±      | 0.158    | ops/s |
| testLongChineseMessageBody  | thrpt | 80  | 5.192 ±      | 0.027    | ops/s |
| testShortAsciiMessageBody   | thrpt | 80  | 116843.074 ± | 1220.330 | ops/s |
| testShortChineseMessageBody | thrpt | 80  | 155858.086 ± | 651.504  | ops/s |

### 0.7.2

| Benchmark                   |  Mode | Cnt | Score        | Error   | Units |
|-----------------------------|-------|-----|--------------|---------|-------|
| testLongAsciiMessageBody    | thrpt | 80  | 1189.459 ±   | 42.306  | ops/s |
| testLongChineseMessageBody  | thrpt | 80  | 3175.659 ±   | 13.284  | ops/s |
| testShortAsciiMessageBody   | thrpt | 80  | 119996.542 ± | 888.424 | ops/s |
| testShortChineseMessageBody | thrpt | 80  | 155339.674 ± | 852.837 | ops/s |

So, in short, the new approach is about the same when messages don't have to be trimmed (which makes sense) and three orders of magnitude faster in some cases when they DO need to be trimmed. In the process of building and running the benchmarks, we DID discover (and fix!) a separate issue that could cause payload generation to hang. See #287 for details.

I think *this* issue is now under control, and will put out a point release to address the hang as soon as possible.
